datanode/datanode.go

```diff
@@ -257,20 +258,43 @@ func (d *dataNode) startDebugServer() {
 	d.opts.InstrumentOptions().Logger().Fatal(http.ListenAndServe(fmt.Sprintf(":%d", d.opts.ServerConfig().DebugPort), debugRouter))
 }
 
+func (d *dataNode) startTableAdditionWatch() {
+	shardOwnershipEvents, done, err := d.metaStore.WatchShardOwnershipEvents()
+	if err != nil {
+		utils.GetLogger().With("error", err.Error()).Fatal("failed to watch schema addition")
+	}
+
+	for {
+		select {
+		case <-d.close:
+			return
+		case event, ok := <-shardOwnershipEvents:
```

Comment:
curious what the second return value from channel read is

Reply:
when channel is closed, ok will be false